### PR TITLE
SOL-701 added the new tab for user preference settings

### PR DESCRIPTION
**Pull request by muhhshoaib**
@chrisndodge Kindly review the changes.

- added the new tab settings for notification preferences.

Thanks

### CODE REVIEW

**Comment by kashifch** [comment]
I will add the bok-choy tests for this tab tomorrow

**Comment by chrisndodge** [comment]
Thanks. I'm looking at it now. There's a lot to go through, so I hope I can complete it by end of day.

File: edx_notifications/server/web/templates/django/notifications_widget.html [hunk 1]
```diff
@@ -11,12 +11,37 @@
...
 <script src="{{ STATIC_URL }}edx_notifications/js/collections/notification_collection.js"></script>
+<script src="{{ STATIC_URL }}edx_notifications/js/collections/notification_preferences_collection.js"></script>
 
 <script src="{{ STATIC_URL }}edx_notifications/js/views/counter_icon_view.js"></script>
 <script src="{{ STATIC_URL }}edx_notifications/js/views/notification_pane_view.js"></script>
+<script src="{{ STATIC_URL }}edx_notifications/js/views/notification_preferences_view.js"></script>
 
-
+<script type="text/template" id="xns-notification-preferences-template">
+    <h2>Notification Preferences</h2>
```
Comment (chrisndodge) [inline edx_notifications/server/web/templates/django/notifications_widget.html]: can you add a div wrapper around this with a class name, so we can apply targeted styling?

**Comment by chrisndodge** [comment]
Thanks. Looks good. I just made a couple of small tweeks (added a few divs and made the defaults configurable).

**Pull request merged**